clients/drcachesim/tools/reuse_distance.h

```diff
@@ -49,13 +49,18 @@
 #include "memref.h"
 
 // We see noticeable overhead in release build with an if() that directly
-// checks knob_verbose, so for debug-only uses we turn it into something the
-// compiler can remove for better performance without going so far as ifdef-ing
-// big code chunks and impairing readability.
+// checks knob_verbose, so for non-debug uses we eliminate it entirely.
+// Example usage:
+//   IF_DEBUG_VERBOSE(1, std::cerr << "This code was executed.\n")
 #ifdef DEBUG
-#    define DEBUG_VERBOSE(level) (reuse_distance_t::knob_verbose >= (level))
+#    define IF_DEBUG_VERBOSE(level, action)                  \
+        do {                                                 \
+            if (reuse_distance_t::knob_verbose >= (level)) { \
```

Comment:
This global seems unnecessary: could pass in `this` instead and then better support multiple instances at once.  Could just add a XXX comment about that for future cleanup.